text/0143-versioning-and-stability.md

```diff
@@ -0,0 +1,128 @@
+
+For the purposes of versioning, all code within a release is treated as if it were part of a single package. All packages within a release receive the same version number. By keeping version numbers in lockstep, end users can understand which packages belong to which release.
+
+Note that contrib packages are released separately from core packages. Contrib releases are not required to match version numbers with the core releases.
+
+In some languages, package managers react poorly to experimental packages having a version higher than 0.X. In these cases, a language-specific workaround is required.
+
+Note: different languages implementations do not need to have matching version numbers, nor do implementations have to match the version of the specification they implement. For example, it is fine to have Python at 1.2.8, Java at 1.3.2, and the spec at 1.1.1. Within python, all packages in the 1.2.8 release have the version number 1.2.8, even if they are experimental.
+
+**Major version bump**  
+Major version bumps only occur when there is a breaking change to a stable interface, or the removal of deprecated signals.
+
+OpenTelemetry values long term support. The expectation is that we will version to v1.0 once the first set of packages are declared stable. OpenTelemetry will then remain at v1.0 for years. THere are no plans for a v2.0 of OpenTelemetry at this time. Additional stable packages, such as metrics and logs, will be added as minor version bumps.
+
+**Minor version bump**  
+Most changes to OpenTelemetry result in a minor version bump.
+
+* New backward-compatible functionality added to any component.
+* Breaking changes to internal SDK components.
+* Breaking changes to experimental signals.
+* New experimental packages are added.
+* Experimental packages become stable.
+
+**Patch version bump**  
+Patch versions make no changes which would require recompilation or potentially break application code. The following are examples of patch fixes. 
+
+* Bugs
+* Security fixes
+* Documentation
+
+Currently, OpenTelemetry does have plans to backport bug and security fixes to prior minor versions. Security and bug fixes are only applied to the latest minor version. We are committed to making it feasible for end users to stay up to date with the latest version of OpenTelemetry.
+
+## Open questions
+
+### “OpenTelemetry GA” aka v1.0
+
+In theory, we have assumed that tracing and metrics would be released together as a v1.0, which we refer to as “OpenTelemetry GA.”
+
+However, in practice, it appears that tracing will be ready to GA before metrics. Tracing is ready today in .NET, and metrics are still months away from being finished.
+
+While we can continue to use the term OpenTelemetry GA to mean the release of both tracing and metrics, we should decouple this from our versioning and support terminology.  That allows us to announce stable tracing this month.
```

Comment:
I like this goal, but I'd love to be more aggressive with it.  We should release components we know to be stable ASAP, and evolve in a compatible way as we go.

As you know, I'd love for us to launch the portion of metrics we consider stable as well, but it may be more work to clearly denote them than it is to just finish up the remaining work on metrics.

Reply:
Without a complete metrics SDK spec, I don't think we can, in good faith, release a metrics implementation. Do you see some way we could?

Reply:
I agree, it is not realistic for a signal to be declared stable until its spec is stable. If the spec can be carved up in such a way that we can declare part of it stable, great.